src/renderer/renderScene.ts

```diff
@@ -351,354 +355,248 @@ const renderLinearElementPointHighlight = (
+  visibleElements.forEach((element) => {
+    // Getting the element using LinearElementEditor during collab mismatches version - being one head of visible elements due to
+    // ShapeCache returns empty hence making sure that we get the
+    // correct element from visible elements
+    if (appState.editingLinearElement?.elementId === element.id) {
+      if (element) {
+        editingLinearElement = element as NonDeleted<ExcalidrawLinearElement>;
+      }
+    }
+  });
 
-          if (
-            frameId &&
-            ((renderConfig.isExporting &&
-              isOnlyExportingSingleFrame(elements)) ||
-              (!renderConfig.isExporting &&
-                appState.frameRendering.enabled &&
-                appState.frameRendering.clip))
-          ) {
-            context.save();
+  if (editingLinearElement) {
+    renderLinearPointHandles(context, appState, editingLinearElement);
+  }
 
-            const frame = getTargetFrame(element, appState);
+  // Paint selection element
+  if (appState.selectionElement) {
+    try {
+      renderSelectionElement(appState.selectionElement, context, appState);
+    } catch (error: any) {
+      console.error(error);
+    }
+  }
 
-            if (frame && isElementInFrame(element, elements, appState)) {
-              frameClip(frame, context, renderConfig);
-            }
-            render();
-            context.restore();
-          } else {
-            render();
-          }
-        } catch (error: any) {
-          console.error(error);
-        }
+  if (appState.isBindingEnabled) {
+    appState.suggestedBindings
+      .filter((binding) => binding != null)
+      .forEach((suggestedBinding) => {
+        renderBindingHighlight(context, appState, suggestedBinding!);
       });
+  }
 
-    if (editingLinearElement) {
-      renderLinearPointHandles(
-        context,
-        appState,
-        renderConfig,
-        editingLinearElement,
-      );
-    }
-
-    // Paint selection element
-    if (appState.selectionElement) {
-      try {
-        renderElement(
-          appState.selectionElement,
-          rc,
-          context,
-          renderConfig,
-          appState,
-        );
-      } catch (error: any) {
-        console.error(error);
-      }
-    }
+  if (appState.frameToHighlight) {
+    renderFrameHighlight(context, appState, appState.frameToHighlight);
+  }
 
-    if (isBindingEnabled(appState)) {
-      appState.suggestedBindings
-        .filter((binding) => binding != null)
-        .forEach((suggestedBinding) => {
-          renderBindingHighlight(context, renderConfig, suggestedBinding!);
-        });
-    }
+  if (appState.elementsToHighlight) {
+    renderElementsBoxHighlight(context, appState, appState.elementsToHighlight);
+  }
 
-    if (appState.frameToHighlight) {
-      renderFrameHighlight(context, renderConfig, appState.frameToHighlight);
-    }
+  const locallySelectedElements = getSelectedElements(elements, appState);
```

Comment:
is this legacy or are we not passing in the cached `selectedElements` for a reason?

Reply:
legacy - unless you had a reason not to change this in #6747

Reply:
passing now the selectedElements as props to `InteractiveCanvas`

Reply:
I just didn't want to cause conflicts